src/allocator/memserver_allocator.cpp

```diff
@@ -71,10 +71,15 @@ void nvmm_profile_dump(){MEMSERVER_PROFILE_END(NVMM)
 #include "allocator/NVMM_counters.tbl"
                                  MEMSERVER_DUMP_PROFILE_SUMMARY(NVMM)}
 
-Memserver_Allocator::Memserver_Allocator() {
+Memserver_Allocator::Memserver_Allocator(const char *fam_path = "") {
     MEMSERVER_PROFILE_INIT(NVMM)
     MEMSERVER_PROFILE_START_TIME(NVMM)
-    StartNVMM();
+
+    if (fam_path != (char *)strdup(""))
```

Comment:
I think you want to perform a string compare and not just address comparison.
Use "strcmp" instead of just "!="

Also, you might want to add a null check for fam_path before doing strcmp, just to make sure that Memserver_Allocator is called with fam_path as "NULL" and not an empty string.